docroot/profiles/humsci/su_humsci_profile/su_humsci_profile.install

```diff
@@ -25,308 +29,444 @@ function su_humsci_profile_final_task() {
 }
 
 /**
- * Indicates version of profile.
+ * Implements hook_update_last_removed().
  */
-function su_humsci_profile_update_8200() {
-  // Intentionally left empty.
+function su_humsci_profile_update_last_removed() {
+  return 9604;
 }
 
 /**
- * Set the permissions for the hero overlay field.
+ * Create new entities to migrate ECKs.
  */
-function su_humsci_profile_update_8201() {
-  $perms = [
-    'view field_hs_hero_overlay_color',
-    'view own field_hs_hero_overlay_color',
-  ];
-  user_role_grant_permissions(UserInterface::ANONYMOUS_ROLE, $perms);
-  $perms = [
-    'create field_hs_hero_overlay_color',
-    'edit field_hs_hero_overlay_color',
-    'edit own field_hs_hero_overlay_color',
-    'view field_hs_hero_overlay_color',
-    'view own field_hs_hero_overlay_color',
-  ];
-  user_role_grant_permissions(UserInterface::AUTHENTICATED_ROLE, $perms);
-}
-
-/**
- * Enable new themes.
- */
-function su_humsci_profile_update_8202() {
-  \Drupal::service('theme_installer')->install([
-    'humsci_basic',
-    'humsci_traditional',
-    'humsci_colorful',
-  ]);
-  drupal_flush_all_caches();
+function su_humsci_profile_update_9700() {
+  /** @var \Drupal\Core\Entity\EntityFieldManagerInterface $entity_field_manager */
+  $entity_field_manager = \Drupal::service('entity_field.manager');
+  $field_map = $entity_field_manager->getFieldMap();
+
+  // Go through each of the ECK types and bundles and create new HS Entity
+  // bundles with the matching fields, form and display settings from the ECK.
+  foreach (_su_humsci_profile_update_eck_info() as $eck_name => $bundles) {
+    $eck_type = \Drupal::config("eck.eck_entity_type.$eck_name");
+    foreach ($bundles as $bundle_id => $new_bundle_id) {
+      $eck_bundle = \Drupal::config("eck.eck_type.$eck_name.$bundle_id");
+
+      $eck_form = EntityFormDisplay::load("$eck_name.$bundle_id.default");
+      $eck_display = EntityViewDisplay::load("$eck_name.$bundle_id.default");
```

Comment:
This looks like it's only going to clone the default EntityViewDisplay and EntityFormDisplay for each bundle. Do any ECK bundles have additional View or Form displays we'd have to account for?

Reply:
It's unlikely. I don't believe they use view displays to the extent that you and I do.